chore: remove cmake minium version deprecation warning

rickstaa opened the pull request:
Not sure if this is desirable given back-compatibility, but this pull request ensures that the cmake minimum version < 3.5 deprecation warning is no longer thrown:

```bash
Warnings   << catkin_tools_prebuild:cmake /home/ricks/development/work/franka_ros_ws/logs/catkin_tools_prebuild/build.cmake.000.log                                                                        
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):                                                                                                                                    
  Compatibility with CMake < 3.5 will be removed from a future version of                                                                                                                                  
  CMake.                                                                                                                                                                                                   
                                                                                                                                                                                                           
  Update the VERSION argument <min> value or use a ...<max> suffix to tell                                                                                                                                 
  CMake that the project does not need compatibility with older versions. 
```

rickstaa commented on franka_ros/CMakeLists.txt:
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1.3)
+cmake_minimum_required(VERSION 3.5)
```
I'm curious why the `franka_ros` minimum cmake version was kept lower than the submodule versions 🤔.

rickstaa replied:
I now know this is because specific requirements exist for the cmake version used in a meta package (see https://wiki.ros.org/catkin/package.xml#Metapackages).

rickstaa commented:
We can also update to version 3.10 since this is what is suggested by [the migration guide](https://wiki.ros.org/noetic/Migration#Increase_required_CMake_version_to_avoid_author_warning).

rickstaa commented:
@Maverobot, @marcbone I just fixed the latest CI errors, and I think we are now suitable for a merge 👍🏻.

rickstaa commented:
I rebased this PR onto the develop branch and added a CHANGELOG entry.

gollth commented on franka_ros/CMakeLists.txt:
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1.3)
+cmake_minimum_required(VERSION 2.8.3)
```
Why is this not updated to 3.10 as well?

rickstaa replied:
Hey @gollth, thanks for reviewing my PR! I got an error when I updated it to 3.10 since the ROS meta package specs currently require version 2.8.3 (see https://wiki.ros.org/catkin/package.xml#Metapackages and https://github.com/ros-planning/moveit/issues/1619). 

#### How to reproduce

1. Checkout https://github.com/frankaemika/franka_ros/commit/22627eff64b2f186dc713ab214a359d78a859fcb#diff-64225d9dae606e899687341a27336f3cfff2889d702f52095a148a4e6d50cb24 or change the CMake version in the `franka_ros/CMakeLists.txt` file to `VERSION 3.10`.
2. Use `catkin_make` to build the package. 
3. Be greeted by the following error:

```bash
WARNING: The CMakeLists.txt of the metapackage 'franka_ros' contains non standard content. Use the content of the following file instead: /home/ricks/development/work/franka_ros_ws/build/catkin_generated/metapackages/franka_ros/CMakeLists.txt
``` 

> [!NOTE]\
> Catkin-tools doesn't have this problem.

gollth commented on franka_gazebo/models/ambient_sun/model.config:
Is this file necessary for the cmake update?

rickstaa replied:
Good catch! Those should not be there. They were accidentally injected when I was testing why the model had so many shadows (see https://github.com/frankaemika/franka_ros/issues/361).

gollth commented on franka_gazebo/models/ambient_sun/model.sdf:
Is this file necessary for the cmake update?

gollth reviewed (changes requested):
Generally looks good, but if possible remove the additions to `models/`

rickstaa committed:
chore: remove cmake minium version deprecation warning

This commit ensure that the cmake minimum version < 3.5 deprecation
warning is no longer thrown. The minimum cmake version was updated
to 3.10 as this is the version that is recommended by the
[ROS noetic migration guide](https://wiki.ros.org/noetic/Migration#Increase_required_CMake_version_to_avoid_author_warning).
The meta package cmake version has to be kept on 2.8.3 because of some
ROS requirements (see
https://wiki.ros.org/catkin/package.xml#Metapackages).
